### DISCUSSION TITLE
pip-compile support for the bazel integration

### DISCUSSION THREAD

**Discussion by ns-ggeorgiev** [renovatebot/renovate#24821] [Suggest an Idea]
### What would you like Renovate to be able to do?

The current pip-compile is not aware of option of generating the requirements.txt. as result it suggests changes like this:
<img width="602" alt="image" src="https://user-images.githubusercontent.com/92812194/175989238-ef84e2fe-59d0-485a-9266-465f54fc4650.png">

Of course, I can workaround the problem with creating different file naming convention, but I would prefer if this is actually handled, properly from the pip-compile manager.

### If you have any ideas on how this should be implemented, please tell us here.

The parsing mechanism should be enhanced to match the generated from bazel command. In such case it either should run the bazel command, or pass the default arguments bazel uses , fix the command reference as well.

### Is this a feature you are interested in implementing yourself?

Yes

**Comment by lalten** [comment]
Any progress on this one?

Note that rules_python's compile_pip_requirements can upgrade to the same newest versions in the requirements lock file by adding `--upgrade` like so:
```Starlark
compile_pip_requirements(
    name = "requirements",
    extra_args = ["--upgrade"],
)
```
So it would really be enough for Renovate to run `bazel run //:requirements_update` when outdated versions are found.
